src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -482,22 +535,6 @@ private async Task<Project> UpdateProjectInfoAsync(Project project, Checksum inf
                     project = addDocuments(project.Solution, lazyDocumentsToAdd.ToImmutable()).GetProject(project.Id)!;
                 }
 
-                // changed document
```

Comment:
moved 'changed' to the bottom, after we process adds/removes (matches what we do with projects).